src/Microsoft.AspNetCore.Hosting/Internal/WebHost.cs

```diff
@@ -296,7 +296,10 @@ private void EnsureServer()
             }
 
             // Fire the IHostedService.Stop
-            _hostedServiceExecutor?.Stop();
+            if (_hostedServiceExecutor != null)
```

Comment:
Do you need this check at all? You resolved _hostedServiceExecutor  as a required service so I don't think it can be null. Also, you don't check when you call StartAsync.